files/files_test.go

```diff
@@ -0,0 +1,84 @@
+package files
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/Flaque/filet"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFile(t *testing.T) {
+	defer filet.CleanUp(t)
+	dir := filet.TmpDir(t, "")
+	path := filepath.Join(dir, "greeting")
+
+	file := File{
+		Description: "greeting",
+		Path:        path,
+		Content:     "SGVsbG8sIFdvcmxk",
+		Encoding:    "base64",
+		Format:      "file",
+	}
+
+	err := file.extract()
+	if assert.NoError(t, err) {
+		bytes, err := ioutil.ReadFile(path)
+		if assert.NoError(t, err) {
+			assert.Equal(t, "Hello, World", string(bytes))
+		}
+	}
+}
+
+func TestFileSubdir(t *testing.T) {
+	defer filet.CleanUp(t)
+	dir := filet.TmpDir(t, "")
+	path := filepath.Join(dir, "path", "to", "greeting")
+
+	file := File{
+		Description: "greeting",
+		Path:        path,
+		Content:     "SGVsbG8sIFdvcmxk",
+		Encoding:    "base64",
+		Format:      "file",
+	}
+
+	err := file.extract()
+	if assert.NoError(t, err) {
+		bytes, err := ioutil.ReadFile(path)
+		if assert.NoError(t, err) {
+			assert.Equal(t, "Hello, World", string(bytes))
+		}
+	}
+}
+
+func TestZipSubdir(t *testing.T) {
+	defer filet.CleanUp(t)
+	dir := filet.TmpDir(t, "")
+	path := filepath.Join(dir, "path", "to", "unpack")
+
+	file := File{
+		Description: "stuff",
+		Path:        path,
+		Content:     `UEsDBAoAAAAAACyz8063xRQTBAAAAAQAAAADABwAYnllVVQJAANTQzJdU0MyXXV4CwABBOkDAAAE6QMAAGJ5ZQpQSwMECgAAAAAAZrPzTu6a6r0EAAAABAAAAAcAHABkaXIvc3ViVVQJAAO/QzJdv0MyXXV4CwABBOkDAAAE6QMAAHN1YgpQSwMECgAAAAAAKrPzTnp6b+0DAAAAAwAAAAIAHABoaVVUCQADUEMyXVBDMl11eAsAAQTpAwAABOkDAABoaQpQSwECHgMKAAAAAAAss/NOt8UUEwQAAAAEAAAAAwAYAAAAAAABAAAAtIEAAAAAYnllVVQFAANTQzJddXgLAAEE6QMAAATpAwAAUEsBAh4DCgAAAAAAZrPzTu6a6r0EAAAABAAAAAcAGAAAAAAAAQAAALSBQQAAAGRpci9zdWJVVAUAA79DMl11eAsAAQTpAwAABOkDAABQSwECHgMKAAAAAAAqs/NOenpv7QMAAAADAAAAAgAYAAAAAAABAAAAtIGGAAAAaGlVVAUAA1BDMl11eAsAAQTpAwAABOkDAABQSwUGAAAAAAMAAwDeAAAAxQAAAAAA`,
```

Comment:
So this zip file looks like:

```
> hi
> bye
> dir/sub
```

Seems good.

Reply:
Yeah, I'll add a comment to that effect.